### PR TITLE
 Encode and decode cookie value on set and get

### PR DESCRIPTION
**Pull request by re5et**
"The cookie value string can use encodeURIComponent() to ensure that the
string does not contain any commas, semicolons, or whitespace (which are
disallowed in cookie values)." -
https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie

We were seeing cut offs on values that contained commas.

### CODE REVIEW

**Comment by aaronjensen** [comment]
:+1: does this look good @guillaumepotier ?

**Comment by guillaumepotier** [comment]
Seems fair to me. @matghaleb WDYT?

File: build/basil.js [hunk 1]
```diff
@@ -217,8 +217,10 @@
 				// retrieve last updated cookie first
 				for (var i = cookies.length - 1, cookie; i >= 0; i--) {
 					cookie = cookies[i].replace(/^\s*/, '');
-					if (cookie.indexOf(key + '=') === 0)
-						return cookie.substring(key.length + 1, cookie.length);
+					if (cookie.indexOf(key + '=') === 0){
```
Comment (guillaumepotier) [inline build/basil.js]: Why not just having done

```javascript
if (cookie.indexOf(key + '=') === 0)
   return decodeURIComponent (cookie.substring(key.length + 1, cookie.length)); 
```

Would have spared a ligne, some brackets (that breaks CS actually with no space ;) )

**Comment by matghaleb** [comment]
We need to encode the key as well or `Basil.cookie.set('hello;world', 42)` won't work as expected

**Comment by guillaumepotier** [comment]
Okay,

So like @matghaleb pointed out, it would be great to encode key too, and maybe review a bit CS and it would be fine :)

Thx a lot

**Comment by aaronjensen** [comment]
Ok, @matghaleb and @guillaumepotier the key is now encoded and some tests have been added. Anything else?

**Comment by guillaumepotier** [comment]
Aside my last comment you should fix IMHO this is looking good :)

**Comment by aaronjensen** [comment]
Done @guillaumepotier Thanks!

**Comment by guillaumepotier** [comment]
Yes this is great!

Thanks for your awesome contribution!

Best

**Comment by aaronjensen** [comment]
np, looking forward to a release so we can remove the git reference from our package.json :smile: thanks!

**Comment by guillaumepotier** [comment]
Should be good now, released `0.4.2` version. Could you confirm this is ok for you?

Thx

**Comment by aaronjensen** [comment]
Doesn't look like it has made it to npm yet. Maybe it'll take a bit, but letting you know just in case the publish failed.

**Pull request merged**